lib/type.ml

```diff
@@ -159,13 +159,20 @@ and inf_elem ty1 ty2 =  (* ty1 and ty2 should be a TyNumber, TySingleton, TyAtom
      Some (TySingleton (Atom a))
   | (TyTuple tys1, TyTuple tys2) when List.length tys1 = List.length tys2 ->
      List.map2_exn ~f:inf tys1 tys2
-     |> fun tys -> Some (TyTuple tys)
+     |> fun tys ->
+        if List.exists ~f:(fun t -> t = TyBottom) tys then
```

Comment:
nits: It would be nice to define a function like [`sequence`](http://hackage.haskell.org/package/base-4.12.0.0/docs/Prelude.html#v:sequence).
`Type.sequence : Type.t list -> Type.t list option`

Reply:
Is the `sequence` a function to eliminate the bottom type?

Reply:
In my expectation, `sequence` function returns `None` if `TyBottom` exists in the given list. Otherwise, returns `Some list` whose `list` is the given list.

Haskell's `sequence`:

```
❯ ghci
GHCi, version 8.4.3: http://www.haskell.org/ghc/  :? for help
Prelude> sequence ([Just 1, Just 2])
Just [1,2]
Prelude> sequence ([Just 1, Nothing, Just 2])
Nothing
```

My expectation of `Type.sequence`:

```
> Type.sequence ([TyAny; TyUnion [TyNumber; TyAtom]])
Some [TyAny; TyUnion [TyNumber; TyAtom]]
> Type.sequence ([TyAny; TyBottom; TyUnion [TyNumber; TyAtom]])
None
```

Perhaps the name `sequence` is less understandable..

Reply:
Thanks, but the idea of the function `sequence` does not use at this time.